panic while updating an azurerm_linux_virtual_machine_scale_set

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

**terraform 1.0.3**

```hcl
Initializing provider plugins...
- Installing hashicorp/random v3.1.0...
- Installed hashicorp/random v3.1.0 (signed by HashiCorp)
- Installing hashicorp/azurerm v2.70.0...
- Installed hashicorp/azurerm v2.70.0 (signed by HashiCorp)
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_2.70.0`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
resource "azurerm_linux_virtual_machine_scale_set" "scalesetsemarchy" {
  for_each                        = local.semarchy_instance
  name                            = each.value.scaleset_name
  resource_group_name             = azurerm_resource_group.rgenv[each.key].name
  location                        = azurerm_resource_group.rgenv[each.key].location
  instances                       = 1
  sku                             = each.value.sku
  admin_username                  = each.value.adminusername
  admin_password                  = random_password.semarchypasswd[each.key].result
  disable_password_authentication = each.value.disable_password_authentication
  zone_balance                    = each.value.zone_balance
  upgrade_mode                    = each.value.upgrade_mode
  custom_data                     = base64encode(local.templatecloudinitconfigsemarchy)
  zones                           = each.value.zones
  source_image_id                 = var.semarchy_image

  os_disk {
    storage_account_type = each.value.storage_account_type
    disk_size_gb         = each.value.disk_size_gb
    caching              = each.value.caching
  }

  boot_diagnostics {
    storage_account_uri = azurerm_storage_account.storsemarchy[each.key].primary_blob_endpoint
  }

  automatic_instance_repair {
    enabled = true
  }

  automatic_os_upgrade_policy {
    disable_automatic_rollback  = false
    enable_automatic_os_upgrade = true
  }

  network_interface {
    name                          = "nic"
    primary                       = true
    enable_accelerated_networking = true
    ip_configuration {
      name                                   = "internal"
      primary                                = true
      subnet_id                              = azurerm_subnet.this["${each.key}-${local.env}-${local.semarchy_layer}"].id
      load_balancer_backend_address_pool_ids = [azurerm_lb_backend_address_pool.backendpool[each.key].id]
    }
  }

  tags = local.global_tags
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

```hcl
│ Error: Plugin did not respond
│ 
│   with azurerm_linux_virtual_machine_scale_set.scaleset[MASKED]["westeurope"],
│   on 05_[MASKED].tf line 29, in resource "azurerm_linux_virtual_machine_scale_set" "scaleset[MASKED]":
│   29: resource "azurerm_linux_virtual_machine_scale_set" "scaleset[MASKED]" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-azurerm_v2.70.0_x5 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4910b4d]
...
```

### Expected Behaviour

<!--- What should have happened? --->

Just applying the expected plan [terraform plan](https://gist.github.com/beparmentier/287e828509ffd0b551f79a6958cd1c66)

### Actual Behaviour

<!--- What actually happened? --->

[terraform panic](https://gist.github.com/beparmentier/59be18203e4d337d88f33e3d19092f63)

terraform crash with a panic

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

I don't really know...

I don't get any issue while creating the scaleset.

I just update the `azurerm_linux_virtual_machine_scale_set` to add

```hcl
  automatic_instance_repair {
    enabled = true
  }

  automatic_os_upgrade_policy {
    disable_automatic_rollback  = false
    enable_automatic_os_upgrade = true
  }
```

and I get this error

I also got this issue while updating the `source_image_id`.

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

Running terraform in Azure West Europe

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/azurerm/internal/services/compute/virtual_machine_scale_set_update.go
+++ b/azurerm/internal/services/compute/virtual_machine_scale_set_update.go
@@ -4,6 +4,7 @@ import (
 	"context"
 	"fmt"
 	"log"
+	"strings"
 
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-12-01/compute"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/client"
@@ -49,7 +50,7 @@ func (metadata virtualMachineScaleSetUpdateMetaData) performUpdate(ctx context.C
 		if userWantsToRollInstances {
 			// If the updated image version is not "latest" and upgrade mode is automatic then azure will roll the instances automatically.
 			// Calling upgradeInstancesForAutomaticUpgradePolicy() in this case will cause an error.
-			if upgradeMode == compute.Automatic && *update.VirtualMachineProfile.StorageProfile.ImageReference.Version == "latest" {
+			if upgradeMode == compute.Automatic && isUsingLatestImage(update) {
 				if err := metadata.upgradeInstancesForAutomaticUpgradePolicy(ctx); err != nil {
 					return err
 				}
@@ -184,4 +185,16 @@ func (metadata virtualMachineScaleSetUpdateMetaData) upgradeInstancesForManualUp
 
 	log.Printf("[DEBUG] Rolled the VM Instances for %s Virtual Machine Scale Set %q (Resource Group %q).", metadata.OSType, id.Name, id.ResourceGroup)
 	return nil
+}
+
+func isUsingLatestImage(update compute.VirtualMachineScaleSetUpdate) bool {
+	if update.VirtualMachineProfile.StorageProfile == nil ||
+		update.VirtualMachineProfile.StorageProfile.ImageReference == nil ||
+		update.VirtualMachineProfile.StorageProfile.ImageReference.Version == nil {
+		return false
+	}
+	if strings.EqualFold(*update.VirtualMachineProfile.StorageProfile.ImageReference.Version, "latest") {
+		return true
+	}
+	return false
 }

